the-collab-lab/tcl-6-smart-shopping-list
# Detail view design

## Type of Changes



|     | Type                       |
| --- | -------------------------- |
|     | :bug: Bug fix              |
|     | :sparkles: New feature     |
| ✓   | :sunglasses: Styling       |
|     | :hammer: Refactoring       |
|     | :100: Add tests            |
|     | :link: Update dependencies |
|     | :scroll: Docs              |

## Related Issue



closes #18 

## Description


The UI from clicking a button to view an item's details to the display of that item's details is cohesive with the rest of the app's styling.

We incorporated a modal view for the item details.

## List To Complete Task

- [x] Fix bugs in the modal display and routing.
- [x] Add color theme to header/footer.
- [x] Style the buttons appropriately.

## Updates

### Before
WITHOUT MODAL:
![Screen Shot 2020-05-25 at 12 30 53 PM](https://user-images.githubusercontent.com/7929227/82840447-7f325500-9e87-11ea-9ea6-6009091c6eda.png)


### After
<img width="1169" alt="Screen Shot 2020-05-26 at 11 54 32 AM" src="https://user-images.githubusercontent.com/48725513/83046820-61d2c780-9ffc-11ea-84e6-35dd77f23741.png">


## Acceptance Criteria



- The item detail view is aligned with the app's theme.

## Testing Steps / QA Criteria



View the deployed branch [here](https://deploy-preview-48--tcl-6-smart-shopping-list.netlify.app/).
Join a shopping list (`stamp avow boil` is a valid token) or view your own shopping list.
Click <some HTML element> (As of my starting this PR it is the "View Details" link on an item).
A modal with the relevant details for the item is displayed and looks good.
